jujugui/static/gui/src/app/components/panel/test-panel.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var juju = {components: {}};
+var testUtils = React.addons.TestUtils;
+
+chai.config.includeStack = true;
+chai.config.truncateThreshold = 0;
+
+describe('PanelComponent', function() {
+
+  beforeAll(function(done) {
+    // By loading this file it adds the component to the juju components.
+    YUI().use('panel-component', function() { done(); });
+  });
+
+  it('generates a visible panel if services are provided', function() {
+    var instanceName = 'custom-instance-name';
+    var services = ['']; // one service.
+
+    var shallowRenderer = testUtils.createRenderer();
+    shallowRenderer.render(
+        <juju.components.Panel
+          instanceName={instanceName}
+          services={services}/>);
+
+    var output = shallowRenderer.getRenderOutput();
+    assert.equal(output.props.className, 'panel-component ' + instanceName);
+    assert.deepEqual(output.props.children,
+      <juju.components.AddedServicesList services={services} />);
+  });
+
+  it('generates a hidden panel if no services are provided', function() {
+    var instanceName = 'custom-instance-name';
+    var services = []; // no services.
+
+    var shallowRenderer = testUtils.createRenderer();
+    shallowRenderer.render(
+        <juju.components.Panel
+          instanceName={instanceName}
+          services={services}/>);
+
+    var output = shallowRenderer.getRenderOutput();
+    assert.equal(output.props.className, 'panel-component ' + instanceName + ' hidden');
```

Comment:
`output.props` is contains the props that are defined on the outer element of the component
`output.props.children` are all of the children elements of the component.